### PR TITLE
Assets starts in depot

### PR DESCRIPTION
**Pull request by humleflue**

### CODE REVIEW

File: DataAccessLibrary/DataModels/AssetRecordData.cs [hunk 1]
```diff
@@ -17,19 +17,21 @@ public class AssetRecordData
 
         public AssetRecordData(IAssetRecord assetRecord)
         {
+            if (assetRecord == null)
+            {
+                throw new ArgumentException();
```
Comment (Pattrigue) [inline DataAccessLibrary/DataModels/AssetRecordData.cs:22]: Should probably include a message with these two exceptions (line 30 too)

File: Server/Components/AssetTable/AssetTable.razor.cs [hunk 2]
```diff
@@ -37,10 +35,10 @@ private void FilterMenuShow(MouseEventArgs e)
         protected override async Task OnInitializedAsync()
         {
             await GetAssetsAsync();
-            
+
             navigator = new PageNavigator<IAsset>(assets, out pageAssets, AssetsPerPage);
             navigator.PageChanged += GetPageAssets;
-            AssetService.AssetUpdated += OnAssetUpdated;
+            new AssetService().AssetUpdated += OnAssetUpdated;
```
Comment (Pattrigue) [inline Server/Components/AssetTable/AssetTable.razor.cs:41]: AssertService is a Singleton and should not be instantiated!

**Review by humleflue — COMMENTED** [review]
Screenshot:
image

File: Server/Pages/AssetDetails.razor [hunk 3]
```diff
@@ -7,7 +7,8 @@
 
 @if (asset == null)
 {
-    <LoadingBar />
+    @*<LoadingBar />*@
+    <p>Loading ...</p>
```
Comment (Pattrigue) [inline Server/Pages/AssetDetails.razor:11]: Please do not remove the LoadingBar component

**Review by Ivikhostrup — APPROVED** [review]
Looks good to me.

**Pull request merged**